### ISSUE TITLE
[P1] Expand training metrics — raw loss, EMA, rolling mean, task-level loss, gradient norm

### ISSUE THREAD

**Issue by fritzprix** [fritzprix/systemone-lite#10]
# Issue: Expand training metrics — raw loss, EMA, rolling mean, task-level loss, gradient norm

## Problem

Current run `spatial-v2-s1-cloze` only logged:
- `train/avg_loss` (likely cumulative average)
- `val/accuracy` (aggregate, single number)
- `train/lr`

This made it impossible to detect that training quality **degraded** after resume:
- Raw loss jumped from 0.875 (25-30k) to 1.302 (40-45k) — but cumulative average masked this
- The 40k val spike of 56% looked promising, but raw loss was worsening
- No per-task breakdown meant connect4 gains masked sokoban losses

## Postmortem Findings

The cumulative average of `train/avg_loss` naturally flattens even when recent loss is increasing. This gave a false impression of recovery.

### 1. Add to W&B logging

```python
# Per-batch metrics:
wandb.log({
    "train/loss_raw": raw_batch_loss,
    "train/loss_ema_100": ema_loss(window=100),
    "train/loss_rolling_500": rolling_mean(window=500),
    "train/loss_by_task/" + task: task_loss,  # one per task
    "train/gradient_norm": gradient_norm,
    "train/update_norm": update_norm,
    "train/tokens_seen": tokens_seen,
    "train/examples_seen": examples_seen,
    "train/optimizer_step": optimizer_step,
    "train/effective_batch_size": physical_batch * grad_accum,
})
```

### 2. Validation metrics

```python
# Per-validation:
wandb.log({
    "val/accuracy": overall_acc,
    "val/macro_accuracy": macro_avg_acc,
    "val/accuracy/" + task: task_acc,  # one per task
    "val/examples_seen": val_samples,
})

# Store per-example predictions for later analysis
# (optional, may need artifact upload due to size)
```

### 3. Analysis tooling

Add a script to compute:
- McNemar test between checkpoints
- Paired bootstrap confidence intervals
- Per-task delta analysis

## Acceptance Criteria

- [ ] `train/loss_raw` logged per batch
- [ ] `train/loss_ema_100` logged every 100 optimizer steps
- [ ] `train/loss_rolling_500` logged every 500 optimizer steps
- [ ] `train/loss_by_task/*` logged per batch
- [ ] `train/gradient_norm` logged per optimizer step
- [ ] `val/macro_accuracy` logged every validation
- [ ] `val/accuracy/*` logged per task every validation
- [ ] `scripts/analyze_checkpoints.py` or similar tool for McNemar/paired tests
